scripts/Muon/GUI/Common/load_run_widget/presenter.py

```diff
@@ -0,0 +1,320 @@
+        self._load_thread.start()
+
+    def handle_load_thread_finished_current_run(self):
+
+        self._load_thread.deleteLater()
+        self._load_thread = None
+
+        self.on_loading_current_run_finished()
+
+    def on_loading_current_run_finished(self):
+        # If in single file mode, remove the previous run
+        if not self._load_multiple_runs and len(self._model.loaded_runs) > 1:
+            self._model.remove_previous_data()
+
+        # if loaded successfully
+        if self._model.loaded_runs:
+            current_run = self._model.loaded_runs[0]
+            self._view.set_run_edit_without_validator(str(current_run) + " (CURRENT RUN)")
+            self._model.current_run = current_run
+        self._view.notify_loading_finished()
+        self.enable_loading()
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Loading from increment/decrement run buttons
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def handle_increment_run(self):
+        run_list = self.get_incremented_run_list()
+        if not run_list:
+            return
+        new_run = max(run_list)
+
+        if self._model.current_run and new_run > self._model.current_run:
+            self._view.warning_popup("Requested run exceeds the current run for this instrument")
+            return
+
+        file_name = file_utils.file_path_for_instrument_and_run(self.get_current_instrument(), new_run)
+        self.handle_loading([file_name], self._use_threading)
+
+    def handle_decrement_run(self):
+        run_list = self.get_decremented_run_list()
+        if not run_list:
+            return
+        new_run = min(run_list)
+
+        file_name = file_utils.file_path_for_instrument_and_run(self.get_current_instrument(), new_run)
+        self.handle_loading([file_name], self._use_threading)
+
+    def get_incremented_run_list(self):
+        """
+        Updates list of runs by adding a run equal to 1 after to the highest run.
+        """
+        run_list = self.flatten_run_list(copy.copy(self.runs))
+        if run_list is None or len(run_list) == 0:
+            return []
+        if len(run_list) == 1:
+            run_list = [run_list[0], run_utils.increment_run(run_list[0])]
+        else:
+            run_list = run_utils.increment_run_list(run_list)
+        return run_list
+
+    def get_decremented_run_list(self):
+        """
+        Updates list of runs by adding a run equal to 1 before to the lowest run.
+        """
+        run_list = self.flatten_run_list(copy.copy(self.runs))
+        if run_list is None or len(run_list) == 0:
+            return []
+        if len(run_list) == 1:
+            run_list = [run_utils.decrement_run(run_list[0]), run_list[0]]
+        else:
+            run_list = run_utils.decrement_run_list(run_list)
+        return run_list
```

Comment:
I have noticed that the start threading stuff has two very similar functions. Is there a way we could reduce them to a single function (with appropriate ifs)?